src/cryptonote_core/service_node_rules.cpp

```diff
@@ -14,15 +15,73 @@ uint64_t get_staking_requirement(cryptonote::network_type m_nettype, uint64_t he
   if (m_nettype == cryptonote::TESTNET || m_nettype == cryptonote::FAKECHAIN)
       return COIN * 100;
 
+  if (hf_version >= cryptonote::network_version_13_enforce_checkpoints)
+  {
+    constexpr uint64_t y[] = {
+        385824,
+        429024,
+        472224,
+        515424,
+        558624,
+        601824,
+        645024,
+        688224,
+        731424,
+        774624,
+        817824,
+        861024,
+        1000000,
+    };
+
+    constexpr uint64_t x[] = {
```

Comment:
x and y array names are swapped from what they should be (I named them x and y for the x-axis being the block height, and the y-axis being the lsr).  Perhaps it would be clearer to rename the heights `heights` and the requirements `lsr`, so the equation is:

```C++
uint64_t result = lsr[i] + (height - heights[i]) * ((lsr[i + 1] - lsr[i]) / (heights[i + 1] - heights[i]));
```

(but given my comment below about the negative value this also either needs everything to be `int64_t`s, or else rewritten as the all-positive:

```C++
uint64_t result = lsr[i] - (height - heights[i]) * ((lsr[i] - lsr[i + 1]) / (heights[i + 1] - heights[i]));
```
)

Reply:
Fixed- added some unit tests for verification